jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/authentication/DigestAuthenticator.java

```diff
@@ -300,18 +312,39 @@ private static class Digest extends Credential
         {
             method = m;
         }
+        
+        Digest(String m, Algorithm a)
+        {
+            method = m;
+            algorithm = a;
+        }
 
         @Override
         public boolean check(Object credentials)
+        {
+            byte[] digest = new byte[]{};
+            digest = calcDigest(credentials, algorithm.getName());
+
+            // check digest
+            return stringEquals(TypeUtil.toString(digest, 16).toLowerCase(), response == null ? null : response.toLowerCase());
+        }
+
+        @Override
+        public String toString()
+        {
+            return username + "," + response;
+        }
+        
+        private byte[] calcDigest(Object credentials, String algorithm)
         {
             if (credentials instanceof char[])
                 credentials = new String((char[])credentials);
             String password = (credentials instanceof String) ? (String)credentials : credentials.toString();
-
+            
             try
             {
                 // MD5 required by the specification
-                MessageDigest md = MessageDigest.getInstance("MD5");
+                MessageDigest md = MessageDigest.getInstance(algorithm);
```

Comment:
Call `getAlgorithm()` here.

Reply:
`Digest` class is static one, `getAlgorithm()` is non-static.
Do you want to have getter for algorithm field in `Digest` class?

Reply:
That line should be:

```suggestion
                MessageDigest md = MessageDigest.getInstance(getAlgorithm());
```

So yes, add `DigestAuthenticator.getAlgorithm()`.

Reply:
OK, fixed!